### PR TITLE
Email opt in

### PR DESCRIPTION
**Pull request by grogsy**
Add a feature  that allows the user to choose to opt-in to the app's weekly email reports. By default user's are opted out, but the user can head to the settings page and clicking on the button under the company panel to opt in to weekly emails. The user can also click the same button to opt out if they wish to no longer receive weekly emails.
Untitled

### CODE REVIEW

**Comment by jt5301** [comment]
i think the  "update email address" button should be changed to something else, since it's more like a subscribe button now?

**Comment by grogsy** [comment]
> 
> 
> i think the "update email address" button should be changed to something else, since it's more like a subscribe button now?

Ah I added back the update email address button, The button next to it is a subscribe/unsubscribe button. If the user isn't opt in for emails the "Opt Out Of Weekly Emails" button will suggest for the user to subscribe.

File: server/models/user.js [hunk 1]
```diff
@@ -24,6 +24,10 @@ const User = db.define("User", {
       return () => this.getDataValue("password");
     },
   },
+  emailOptIn: {
+    type: DataTypes.BOOLEAN,
+    defaultValue: false,
+  },
 });
```
Comment (jt5301) [inline server/models/user.js:31]: The email scraper is set to just send emails to everyone that has a subscriber email, so right now this boolean isn't really indicating anything on that file. Can you add a check to the emailscraper file to only send to users that have emailOptIn to be true? Let me know if you need help reading my code! I can also do it too if anything.

Comment (grogsy) [reply server/models/user.js:31]: Sure John I can do that. I just wasn't sure if you wanted to be the one to do it so I left it out for now

Comment (jt5301) [reply server/models/user.js:31]: Looks good to me, yeah I didn't mind either way, TY!

**Review by jt5301 — APPROVED** [review]

**Pull request merged**